File: Sum of square numbers/SquareSum.cpp
class Solution {
public:
    bool judgeSquareSum(int c) {
        for (int prime = 2; prime <= std::sqrt(c); ++prime) {
            if (c % prime == 0) {
                int count = 0;
                while (c % prime == 0) {
                    c /= prime;
                    ++count;
                }
                if (prime % 4 == 3 && count % 2 != 0) {
                    return false;
                }
            }
        }
        return c % 4 != 3;
    }
};

File: Sum of square numbers/SquareSum.py
class Solution:
    def judgeSquareSum(self, c: int) -> bool:
        for prime in range(2, int(math.sqrt(c)) + 1):
            if c % prime == 0:
                count = 0
                while c % prime == 0:
                    c //= prime
                    count += 1
                if prime % 4 == 3 and count % 2 != 0:
                    return False
        return c % 4 != 3

        